indico/util/user.py

```diff
@@ -26,112 +25,30 @@ def iter_acl(acl):
-                            address=data.get('address', ''), phone=data.get('phone', ''),
-                            affiliation=data.get('affiliation', ''), is_pending=True)
-                db.session.add(user)
-                db.session.flush()
-        else:
-            raise ValueError(f"Id '{id_}' is not a number and allow_pending=False")
-        if user is None:
-            raise ValueError(f'User does not exist: {id_}')
-        return user
-    elif allow_emails and type_ == 'Email':
-        return EmailPrincipal(id_)
-    elif allow_networks and type_ == 'IPNetworkGroup':
-        group = IPNetworkGroup.get(int(id_))
-        if group is None or (group.hidden and group not in existing_data):
-            raise ValueError(f'IP network group does not exist: {id_}')
-        return group
-    elif allow_groups and type_ in {'LocalGroupWrapper', 'LocalGroup'}:
-        group = GroupProxy(int(id_))
-        if group.group is None:
-            raise ValueError(f'Local group does not exist: {id_}')
-        return group
-    elif allow_groups and type_ in {'LDAPGroupWrapper', 'MultipassGroup'}:
-        provider = fossil['provider']
-        group = GroupProxy(id_, provider)
-        if group.group is None and not allow_missing_groups:
-            raise ValueError(f'Multipass group does not exist: {provider}:{id_}')
-        return group
-    elif category and type_ == 'CategoryRole':
-        role = CategoryRole.get_category_role_by_id(category, id_)
-        role_name = fossil.get('name')
-        if role is None:
-            raise ValueError(f'Category role "{role_name}" is not available in "{category.title}"')
-        return role
-    elif event and type_ == 'EventRole':
-        role = EventRole.get(id_)
-        role_name = fossil.get('name')
-        if role is None:
-            raise ValueError(f'Event role "{role_name}" does not exist')
-        if role.event != event:
-            raise ValueError(f'Event role "{role_name}" does not belong to "{event.title}"')
-        return role
-    elif allow_registration_forms and type_ == 'RegistrationForm':
-        registration_form = RegistrationForm.get(id_)
-        reg_form_name = fossil.get('title')
-        if registration_form is None:
-            raise ValueError(f'Registration form "{reg_form_name}" does not exist')
-        if registration_form.event != event:
-            raise ValueError(f'Registration form "{reg_form_name}" does not belong to "{event.title}"')
-        return registration_form
-    else:
-        raise ValueError(f'Unexpected fossil type: {type_}')
-
-
 def principal_from_identifier(identifier, allow_groups=False, allow_external_users=False, allow_event_roles=False,
                               allow_category_roles=False, allow_registration_forms=False, allow_emails=False,
-                              event_id=None, soft_fail=False):
+                              allow_networks=False, event_id=None, category_id=None, soft_fail=False):
```

Comment:
It was there before, and it does apply to everything where it's appropriate:

- don't fail if user is soft-deleted
- don't fail if (remote) group no longer exists
- don't fail if regform is soft-deleted
- category roles that are not available in the current context (because the category/event using it was moved)

this is used when resolving an identifier to details for displaying an ACL in the new react-based widgets, because failing loudly would be disruptive there, while we do reject such items when actually saving (so viewing an ACL still works, but you cannot save an ACL containing invalid stuff)

Reply:
I was focused on newly added part for `Category` and `Event` since those `raise` when `None`.
But I can see it now, I got tricked by the collapsed diff view, as I skipped reading this part in my editor.